kcbq-api/src/main/java/com/wepay/kafka/connect/bigquery/api/TopicAndRecordName.java

```diff
@@ -0,0 +1,70 @@
+package com.wepay.kafka.connect.bigquery.api;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public class TopicAndRecordName {
+  private final String topic;
+  private final String recordName;
+
+  public static TopicAndRecordName from(String topic) {
+    return new TopicAndRecordName(topic, null);
+  }
+
+  public static TopicAndRecordName from(String topic, String recordName) {
+    return new TopicAndRecordName(topic, recordName);
+  }
+
+  public static TopicAndRecordName from(SinkRecord record, boolean includeRecordName) {
+    String maybeRecordName = includeRecordName ? record.valueSchema().name() : null;
+    return from(record.topic(), maybeRecordName);
+  }
+
+  public TopicAndRecordName(String topic, String recordName) {
+    this.topic = topic;
+    this.recordName = recordName;
+  }
+
+  public String getTopic() {
+    return this.topic;
+  }
+
+  public Optional<String> getRecordName() {
+    return Optional.ofNullable(this.recordName);
+  }
+
+  public String toSubject() {
+    return toSubject(KafkaSchemaRecordType.VALUE);
+  }
+
+  public String toSubject(KafkaSchemaRecordType schemaType) {
+    String subject = topic;
+    if (recordName == null) {
+      subject += "-" + schemaType.toString();
+    } else {
+      subject += "-" + recordName;
+    }
+    return subject;
+  }
+
+  @Override
+  public String toString() {
+    return topic + "/" + recordName;
+  }
+
+  @Override
+  public int hashCode() {
+    return this.topic.hashCode();
```

Comment:
Good catch!
According to contract between hashCode and equal, two instances with the same hashCode might not be equal. But I forgot that in case of a single topic with many records we will end up with HashMaps that has only one bucket... I'm going to include recordName in hashCode.